clang/lib/Basic/Targets/SPIR.h

```diff
@@ -56,9 +56,12 @@ static const unsigned SYCLAddrSpaceMap[] = {
 };
 
 class LLVM_LIBRARY_VISIBILITY SPIRTargetInfo : public TargetInfo {
+private:
+  bool IsWindowsSYCL;
```

Comment:
Typically just inheriting from this type for a WindowsSPIRTargetInfo would be the way to do this.